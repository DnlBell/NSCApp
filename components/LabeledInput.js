import React, { Component } from 'react';
import { View, TextInput, Text } from 'react-native';

class LabeledInput extends Component {

  render() {
    const { label, value, ...props } = this.props;
    const labelStyle = {
      position: 'absolute',
      left: 10,
      top: 0,
      fontSize: 14,
      color: '#000'
    };

    return (
      <View style={{ paddingTop: 18, marginTop: 18 }}>
        <Text style={labelStyle}>
          {label}
        </Text>
        <TextInput
          {...props}
          style={{ height: 32, fontSize: 20, color: '#000', borderBottomWidth: 1, borderBottomColor: '#555', marginHorizontal: 10 }}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
          blurOnSubmit
        />
      </View>
    );
  }
}

const styles = {
  
  inputText: {
    backgroundColor: '#fff',
    marginBottom: 8,
    flex: .9,
    padding: 18,
    color: '#000', 
    borderBottomWidth: 1, 
    borderBottomColor: '#555'
  }
}

export default LabeledInput;