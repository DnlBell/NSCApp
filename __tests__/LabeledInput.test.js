import LabeledInput from '../components/LabeledInput';
import renderer from 'react-test-renderer';
import React from 'react';

it('renders correctly', () => {
  const tree = renderer
    .create(<LabeledInput />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});