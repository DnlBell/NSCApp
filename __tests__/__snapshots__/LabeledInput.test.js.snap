// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 18,
      "paddingTop": 18,
      "width": 732,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#5E5E5E",
        "fontSize": 16,
        "left": 10,
        "position": "absolute",
        "top": 0,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={true}
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderBottomColor": "#5E5E5E",
        "borderBottomWidth": 1,
        "color": "#5E5E5E",
        "flex": 0.9,
        "fontSize": 16,
        "height": 32,
        "marginBottom": 9,
        "padding": 18,
      }
    }
  />
</View>
`;
